sample-macos-app/sample-macos-app/ImageLoading.swift

```diff
@@ -0,0 +1,36 @@
+import Cocoa
+
+public extension NSImageView {
+    private var activeTask: URLSessionDataTask? {
+        get {
+            return objc_getAssociatedObject(self, &AssociatedKey.task) as? URLSessionDataTask
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKey.task, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+   
+    func loadImageByUrl(url: String, completion: @escaping () -> Void) {
+        cancelImageLoading()
+
+        let task = URLSession.shared.dataTask(with: URL(string: url)!) { [weak self] data, d, error in
+            guard let data = data, error == nil else { return }
+            DispatchQueue.main.async {
+                self?.image = NSImage(data: data)
```

Comment:
I know, this is just an example, but `NSImage` initialiser from the data is expensive. You can create the image in the main scope and just assign the value to `self?.image` in the dispatch to the main thread.

Reply:
Fixed